crates/net/rpc/src/admin.rs

```diff
@@ -48,10 +50,11 @@ impl AdminApiServer for AdminApi {
         todo!()
     }
 
-    fn node_info(&self) -> RpcResult<NodeInfo> {
+    async fn node_info(&self) -> RpcResult<NodeInfo> {
         let enr = self.network.local_node_record();
+        let status = self.network.network_status().await.expect("Could not fetch status");
```

Comment:
I would also instead convert the error to an `RpcResult` rather than use `expect` here

I notice that we have helpers such as `ToRpcResult`, but this is only implemented for `reth_interfaces::Error`, which does not currently contain a variant for `Network`. This would require a larger refactor than in the scope of this PR

Reply:
Tracking in #936

Reply:
Yeah the `expect` was definitely making me wince a bit but saw that the `ToRpcResult` wasn't implemented and wasn't sure if it was by design or not. I'll cast this to a custom `jsonrpsee` error for now

Reply:
done in https://github.com/paradigmxyz/reth/pull/923/commits/807f7fc66d4e13551fddde02b556cf01d2dd7dbd